trio_websocket/__init__.py

```diff
@@ -4,18 +4,90 @@
 import ssl
 from functools import partial
 
+from async_generator import async_generator, yield_, asynccontextmanager
 import trio
 import trio.abc
 import wsproto.connection as wsconnection
 import wsproto.events as wsevents
 import wsproto.frame_protocol as wsframeproto
+from yarl import URL
 
 
 __version__ = '0.2.0-dev'
 RECEIVE_BYTES = 4096
 logger = logging.getLogger('trio-websocket')
 
 
+@asynccontextmanager
+@async_generator
```

Comment:
I don't think the `@async_generator` belongs here, try removing it

`@asynccontextmanager` implies a generator that yields a single value

Reply:
According to the docs `async_generator` is required for compatibility with Python 3.5 (which is the minimum version specified in setup.py right now). Even with Python 3.6, contrary to what the docs say, it broke when I removed it.